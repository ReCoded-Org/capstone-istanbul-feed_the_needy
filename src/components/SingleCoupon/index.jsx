import React, { useState } from "react";
import { Typography, Button, Row, Col, Card } from "antd";

const { Title } = Typography;

const SingleCoupon = ({
  coupon,
  deleteSingleCoupon,
  setTotalAmount,
  changeAmount,
  isTesting,
}) => {
  const [amount, setAmount] = useState(1);

  return (
    <div>
      <Row>
        <Col>
          <Card hoverable align="center" className="couponCards">
            <Row>
              <Col>
                {isTesting ? (
                  <button type="button">x</button>
                ) : (
                  <Button
                    className="cardDeleteButton"
                    type="primary"
                    onClick={() => {
                      changeAmount(coupon.amount * amount, "decrement");
                      deleteSingleCoupon(coupon.id);
                    }}
                  >
                    X
                  </Button>
                )}
              </Col>
            </Row>
            <Title>{coupon ? coupon.organizationName : null}</Title>
            <img src={coupon ? coupon.img : null} alt="logo" width="120em" />
            <Title>
              {coupon ? coupon.amount : null}
              <span>â‚º</span>
            </Title>
            {isTesting ? (
              <button type="button">-</button>
            ) : (
              <Button
                className="cardButton"
                type="primary"
                onClick={() => {
                  if (amount > 1) {
                    setAmount(amount - 1);
                    changeAmount(coupon ? coupon.amount : null, "decrement");
                  }
                }}
              >
                -
              </Button>
            )}
            <span className="couponAmount">{amount}</span>
            {isTesting ? (
              <button type="button">+</button>
            ) : (
              <Button
                className="cardButton"
                type="primary"
                onClick={() => {
                  setAmount(amount + 1);
                  changeAmount(coupon ? coupon.amount : null, "increment");
                }}
              >
                +
              </Button>
            )}
          </Card>
        </Col>
      </Row>
    </div>
  );
};

export default SingleCoupon;
