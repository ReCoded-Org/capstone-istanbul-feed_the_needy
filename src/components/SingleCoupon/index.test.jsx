import React from "react";
import renderer from "react-test-renderer";
import SingleCoupon from "./index";

// Fix for TypeError: window.matchMedia is not a function
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

it("SingleCoupon renders correctly", () => {
  const tree = renderer.create(<SingleCoupon isTesting />).toJSON();
  expect(tree).toMatchSnapshot();
});
