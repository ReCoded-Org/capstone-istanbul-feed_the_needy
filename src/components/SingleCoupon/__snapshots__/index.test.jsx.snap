// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleCoupon renders correctly 1`] = `
<div>
  <div
    className="ant-row"
    style={Object {}}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        align="center"
        className="ant-card ant-card-bordered ant-card-hoverable couponCards"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-row"
            style={Object {}}
          >
            <div
              className="ant-col"
              style={Object {}}
            >
              <button
                type="button"
              >
                x
              </button>
            </div>
          </div>
          <h1
            className="ant-typography"
            style={
              Object {
                "WebkitLineClamp": null,
              }
            }
          />
          <img
            alt="logo"
            src={null}
            width="120em"
          />
          <h1
            className="ant-typography"
            style={
              Object {
                "WebkitLineClamp": null,
              }
            }
          >
            <span>
              â‚º
            </span>
          </h1>
          <button
            type="button"
          >
            -
          </button>
          <span
            className="couponAmount"
          >
            1
          </span>
          <button
            type="button"
          >
            +
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
